
---
title: "Deconvolution to cancer and normal plasma based on human normal tissues"
author: "Shicheng Guo"
date: "July 7, 2016"
output: html_document
---

```{r setup,echo=F}  

Coloncancerdatapre<-function(){
  rlt<-list()
  setwd("/home/shg047/oasis/monod/hapinfo/June")
  saminfo<-read.table("/home/shg047/oasis/monod/saminfo.txt",sep="\t")
  #data<-read.table("/oasis/tscc/scratch/shg047/monod/hapinfo/June/monod.mhl.june25.txt",head=T,sep="\t",row.names=1,as.is=T,check.names=F)
  #save(data,file="monod.mhl.june25.RData")
  load("monod.mhl.june25.RData")
  data<-data[,-grep("CTT-|PC-P|PC-T",colnames(data))]
  colnames(data)[grep("STL",colnames(data))]<-as.character(saminfo[match(colnames(data)[grep("STL",colnames(data))],saminfo[,1]),2])
  colnames(data)[grep("WB",colnames(data))]<-"WBC"
  colnames(data)[grep("N37",colnames(data))]<-as.character(saminfo[match(colnames(data)[grep("N37",colnames(data))],saminfo[,1]),2])
  colnames(data)[grep("methylC",colnames(data))]<-"H1"
  colnames(data)[grep("6-T|Colon_Tumor_Primary",colnames(data))]<-"CCT"
  colnames(data)[grep("7-T-",colnames(data))]<-"LCT"
  colnames(data)[grep("6-P-",colnames(data))]<-"CCP"
  colnames(data)[grep("7-P-",colnames(data))]<-"LCP"
  colnames(data)[grep("NC-P-",colnames(data))]<-"NCP"
  data1<-data[,grep("Brain|Stomach|Lung|Heart|Colon|WBC|Intestine|Liver|Esophagus|Kidney|CCP|CCT|LCT|CCTmix|NCP|LCTmix|LCP",colnames(data))]
  data1_ref<-data1[,-(grep("CCP|CCTmix|NCP|LCP|LCTmix",colnames(data1)))]
  base=unlist(lapply(strsplit(colnames(data1_ref),"[.]"),function(x) x[[1]]))
  colnames(data1_ref)<-base
  gsi<-data.frame(GSIMethDecon(data1_ref))
  group<-as.character(unique(gsi$group))
  signatures<-AverageWithGroup(data1_ref)  # QR
  rlt$GSI<-gsi
  rlt$data<-data1
  rlt$ref<-data1_ref
  rlt$signature<-signatures
  return(rlt)
}
GSIMethDecon<-function(data){
  data<-data.matrix(data)
  group=names(table(colnames(data)))
  index=colnames(data)
  gsi<-gmaxgroup<-avebase<-c()
  for(i in 1:nrow(data)){
    gsit<-0
    gmax<-names(which.max(tapply(as.numeric(data[i,]),index,function(x) mean(x,na.rm=T))))
    for(j in 1:length(group)){
      tmp<-(1-10^(mean(data[i,][which(index==group[j])],na.rm=T))/10^(mean(data[i,][which(index==gmax)],,na.rm=T)))/(length(group)-1)
      gsit<-gsit+tmp
    }
    ave<-tapply(data[i,], index, function(x) mean(x,na.rm=T))
    gmaxgroup<-c(gmaxgroup,gmax)
    gsi<-c(gsi,gsit)
    avebase<-rbind(avebase,ave)
  }
  rlt=data.frame(region=rownames(data),group=gmaxgroup,GSI=gsi,avebase)
  return(rlt)
}
AverageWithGroup<-function(data){
  base=unlist(lapply(strsplit(colnames(data),"[.]"),function(x) x[[1]]))
  matrix=apply(data,1,function(x) tapply(x,base,function(x) mean(x,na.rm=T)))
  matrix<-t(matrix)
  rownames(matrix)=rownames(data)
  matrix<-matrix[!rowSums(!is.finite(matrix)),]
  return(matrix)
}
TopGSIByCategory<-function(gsi,top=2,thresHigh=0.3,thresLow=0.1,plotfigure=T,figprefix="tissuespecific"){
  GSIRlt<-c()
  group<-as.character(unique(gsi$group))
  rank<-c(rep(top,length(group)))
  otf1<-paste(figprefix,"boxplot.pdf",sep="")
  otf2<-paste(figprefix,"heatmap.pdf",sep="")
  parnum<-ceiling(sqrt(length(group)))
  pdf(otf1)
  par(mfrow=c(parnum,parnum),oma = c(2,2,2,2) + 0.1,mar = c(2,2,2,2) + 0.1,cex.axis=0.75, cex.lab=0.75)
  for (i in 1:length(group)){
    # select tissue-specific (remove target group<0.2 or non-target group>0.1)
    subset=gsi[which(gsi$group==group[i]),]
    rexclhigh<-which(apply(subset,1,function(x) x[grep(group[i],colnames(gsi))]<0.2))
    xx<-subset[,-grep(group[i],colnames(gsi))]
    rexcllow<-which(apply(xx,1,function(x) any(as.numeric(x[4:length(x)])>0.1)))
    rexcl<-c(rexclhigh,rexcllow)
    subset=subset[-rexcl,]
    subset=subset[order(subset[,3],decreasing=T)[1:rank[i]],]
    GSIRlt<-rbind(GSIRlt,subset)
    if(plotfigure==T){
      zz=subset[which(subset$group==group[i]),]
      boxplot(zz[,4:ncol(zz)],horizontal=T,las=2,col="red")
    }
  }
  dev.off()
  
  if(plotfigure==T){
    HeatMap(data=data.matrix(na.omit(GSIRlt[,4:ncol(GSIRlt)])),phen=gsub("AVE.","",colnames(GSIRlt)[4:ncol(GSIRlt)]),figure=otf2)
  }
  return(GSIRlt)
}
DataExtractionTrim<-function(data){
  data<-data[,-grep("CTT-|PC-P|PC-T",colnames(data))]
  colnames(data)[grep("STL",colnames(data))]<-as.character(saminfo[match(colnames(data)[grep("STL",colnames(data))],saminfo[,1]),2])
  colnames(data)[grep("WB",colnames(data))]<-"WBC"
  colnames(data)[grep("N37",colnames(data))]<-as.character(saminfo[match(colnames(data)[grep("N37",colnames(data))],saminfo[,1]),2])
  colnames(data)[grep("methylC",colnames(data))]<-"H1"
  colnames(data)[grep("6-T|Colon_Tumor_Primary",colnames(data))]<-"CCT"
  colnames(data)[grep("7-T-",colnames(data))]<-"LCT"
  colnames(data)[grep("6-P-",colnames(data))]<-"CCP"
  colnames(data)[grep("7-P-",colnames(data))]<-"LCP"
  colnames(data)[grep("NC-P-",colnames(data))]<-"NCP"
  data1<-data[,grep("Brain|Stomach|Lung|Heart|Colon|CCT|WBC|Liver|Esophagus|Kidney|Intestine|CCP",colnames(data))]
  return(data1)
}
HeatMap<-function(data,phen,figure="heatmap.pdf",cexRow = 0.01,cexCol = 1.2,Colv=T,Rowv=T){
  library("gplots")
  colnames(data)=phen
  colors <- colorpanel(75,"midnightblue","mediumseagreen","yellow") 
  colors <-bluered(75)
  colors <-greenred(75)
  sidecol<-function(x){
    x<-as.numeric(as.factor(x))
    col<-rainbow(length(table(colnames(data))))
    sapply(x,function(x) col[x])
  }
  ColSideColors=sidecol(phen)
  pdf(figure)
  heatmap.2(data,trace="none",cexRow = cexRow,cexCol = cexCol, ColSideColors=ColSideColors,density.info="none",col=colors,Colv=Colv,Rowv=Rowv,keysize=0.9, margins = c(5, 10))
  dev.off()
}
FigurePrepareSimulation<-function(){
  library("ggplot2")
  Fig<-data.frame(seq(0,1,by=0.05),Rlt$out.all[,grep("CCT",colnames(Rlt$out.all))],Rlt$out.all[,grep("WB",colnames(Rlt$out.all))])
  colnames(Fig)<-c("Simulated","CCT","WB")
  Fig<-100*Fig
  c <- ggplot(Fig, aes(Simulated,CCT))
  c <- c + xlab("Simulated contribution (%)")+ylab("Predicted contribution (%)")
  c <- c + stat_smooth(se = TRUE,n=10,size=1.5) + geom_point(size=3)
  c <- c + stat_smooth(se = TRUE,n=10,size=1.5) + geom_point(size=3)
  c <- c + theme_bw()
  c <- c + theme(axis.text=element_text(size=10), axis.title=element_text(size=14,face="bold"))
  ggsave(c,file="coloncancer-deconvolution.simultaion-2.pdf")
  save.image(file="Colon-Deconvolution.RData")
}
ci95<-function(x){
  x<-data.frame(x)
  out<-c()
  for(i in 1:ncol(x)){
    error <- qt(0.975,df=length(x[,i])-1)*sd(x[,i])/sqrt(length(x[,i]))
    m<-round(mean(x[,i]),4)
    d<-round(mean(x[,i])-error,4)
    u<-round(mean(x[,i])+error,4)
    rlt<-paste("mean=",m, ", 95%CI:",d,"-",u,sep="")
    out<-cbind(out,rlt)
  }
  return(out)
}
inputrim<-function(NewRealMatrix,NewSignatures){
rlt<-list()
input<-na.omit(data.frame(NewRealMatrix,NewSignatures))
NewRealMatrix=input[,grep("CP|Virtual",colnames(input))]
NewSignatures=input[,-grep("CP|Virtual",colnames(input))]
rlt$NewRealMatrix=NewRealMatrix
rlt$NewSignatures=NewSignatures
return(rlt)
}
RandomSamplingMean<-function(data,number=round(ncol(data)/2)){
  rlt<-c()
  for(i in 1:ncol(data)){
    VirtualSample<-rowMeans(data[,sample(1:ncol(data),number)],na.rm=T)  
    rlt<-cbind(rlt,VirtualSample)
  }
  colnames(rlt)<-paste("VirtualSample",1:ncol(data),sep="")
  return(rlt)
}

```

```{r}
outputprefix="Coloncancer.plasma.0.05.top100"
library("DeconRNASeq")
library("ggplot2")
setwd("/home/shg047/oasis/monod/hapinfo/June")
# Colon<-Coloncancerdatapre()
# save(Colon,file=paste(outputprefix,".input.RData",sep=""))
load("Coloncancer.plasma.0.05.top100.input.RData")
```

```{r featureSelection,echo=F}

########################################################################
################# remove non-linear MHB regions ########################
########################################################################

linearMHB<-function(x,samlab="CCTmix"){
Rlt<-list()
load("Coloncancer.plasma.0.05.top100.input.RData")
data=Colon$data
cctMix=data[,grep(samlab,colnames(data))]
cctMix=cctMix[,c(1:2,10,3:9,11:ncol(cctMix))]
idx<-unlist(lapply(strsplit(colnames(cctMix),"[.]"),function(x) x[[2]]))
y<-as.numeric(cctMix[1,])
sel<-c()
R<-c()
cctMix<-na.omit(cctMix)
dim(cctMix)
for(i in 1:nrow(cctMix)){
  fit<-summary(lm(as.numeric(cctMix[i,])~as.numeric(idx)))
  R2<-fit$r.squared
  R<-c(R,R2)
  if( ! is.na(R2) & R2>0.7){
  sel<-c(sel,i)
  }  
}
newdata=cctMix[sel,]
dataR<-data.frame(newdata)
Rlt$R2<-R
Rlt$data<-dataR
return(Rlt)
}
#rlt<-linearMHB()
#linearMHBs<-rownames(rlt$data)
#save(linearMHBs,file="Colon.Decov.linear.Var.RData")
```

```{r Deconvolution, echo=T,warning=FALSE}
######################################################################################################
############## feature selection of tissue specific MHBs for deconvolution ########################
######################################################################################################
# select MHBs with linear pattern in mixture
load("Colon.Decov.linear.Var.RData")
data1=Colon$data[match(linearMHBs,rownames(Colon$data)),]
data1_ref=Colon$ref[match(linearMHBs,rownames(Colon$ref)),]
# select tissue specific MHBs
signatures<-AverageWithGroup(data1_ref)  
outputprefix="TopGSIByCategory"
topGSIVar<-300
topgsi<-TopGSIByCategory(na.omit(Colon$GSI),top=topGSIVar,thresHigh=0.3,thresLow=0.1,plotfigure=T,figprefix=outputprefix)
# simulation data and real data
VirtualMatrix<-data.frame((data1[,grep("CCTmix",colnames(data1))]))
RealMatrix<-data.frame((data1[,grep("CCP|LCP|NCP",colnames(data1))]))

###################################################
##### Deconvolution-QR-Simulation-CCT #############
###################################################
DeconData<-data.frame(VirtualMatrix,signatures[match(rownames(VirtualMatrix),rownames(signatures)),])
DeconData<-na.omit(DeconData)
DeconData.tmp<-DeconData[na.omit(match(topgsi[,1],rownames(DeconData))),]
VirtualMatrix=data.frame(DeconData.tmp[,grep("CCTmix",colnames(DeconData))])
Signatures=data.frame(DeconData.tmp[,-grep("CCTmix",colnames(DeconData))])
library(car)
# NewVirtualMatrix=logit(VirtualMatrix)
# NewSignatures=logit(Signatures)
NewVirtualMatrix=(VirtualMatrix)
NewSignatures=(Signatures)
library("DeconRNASeq")
Rlt<-DeconRNASeq(NewVirtualMatrix,NewSignatures,checksig=FALSE,known.prop = F, use.scale = TRUE, fig = TRUE)
acc<-Rlt$out.all
rownames(acc)=colnames(NewVirtualMatrix)
acc
fileout=paste("Colon-mixtureSimulation.Deconvolution.",topGSIVar,"M.txt",sep="")
write.table(acc,file=fileout,col.names=NA,row.names=T,sep="\t",quote=F)

idx<-unlist(lapply(strsplit(rownames(acc),"[.]"),function(x) x[[2]]))
y<-as.numeric(acc[,2])
lm.input<-data.frame(x=as.numeric(idx),y=100*y)
lm.input<-lm.input[order(lm.input$x),]
plot(y~x,data=lm.input,type="o",xlab="Expectation",ylab="Observed")
abline(0, 1,lwd=2,col="red")
dev.off()

plot(1,1)
abline(0, 1)




###################################################
##### Deconvolution-QR-Simulation-LCT #############
###################################################

# VirtualMatrix<-data.frame((data1[,grep("LCTmix",colnames(data1))]))
# DeconData<-data.frame(VirtualMatrix,signatures[match(rownames(VirtualMatrix),rownames(signatures)),])
# DeconData<-na.omit(DeconData)
# DeconData.tmp<-DeconData[na.omit(match(topgsi[,1],rownames(DeconData))),]
# VirtualMatrix=data.frame(DeconData.tmp[,grep("LCTmix",colnames(DeconData))])
# Signatures=data.frame(DeconData.tmp[,-grep("LCTmix",colnames(DeconData))])
# library(car)
# NewVirtualMatrix=logit(VirtualMatrix)
# NewSignatures=logit(Signatures)
# NewVirtualMatrix=(VirtualMatrix)
# NewSignatures=(Signatures)
# library("DeconRNASeq")
# Rlt<-try(DeconRNASeq(NewVirtualMatrix,NewSignatures,checksig=FALSE,known.prop = F, use.scale = TRUE, fig = TRUE))
# acc<-Rlt$out.all
# # rownames(acc)=colnames(NewVirtualMatrix)
# acc
# fileout=paste("Lung-mixtureSimulation.Deconvolution.",topGSIVar,"M.txt",sep="")
# write.table(acc,file=fileout,col.names=NA,row.names=T,sep="\t",quote=F)

###################################################
##### Deconvolution-QR-Real-Plasma-Data #############
###################################################
DeconData<-data.frame(RealMatrix,signatures[match(rownames(RealMatrix),rownames(signatures)),])
DeconData.tmp<-DeconData[na.omit(match(topgsi[,1],rownames(DeconData))),]
RealMatrix1=data.frame(DeconData.tmp[,grep("CCP",colnames(DeconData))])
RealMatrix2=data.frame(DeconData.tmp[,grep("NCP",colnames(DeconData))])
RealMatrix3=data.frame(DeconData.tmp[,grep("LCP",colnames(DeconData))])
Signatures=data.frame(DeconData.tmp[,-grep("CCP|LCP|NCP",colnames(DeconData))])
# Normalization (logit transform)
library(car)
# NewRealMatrix1=logit(RealMatrix1)
# NewRealMatrix2=logit(RealMatrix2)
# NewRealMatrix3=logit(RealMatrix3)
NewRealMatrix1=(RealMatrix1)
NewRealMatrix2=(RealMatrix2)
NewRealMatrix3=(RealMatrix3)
NewSignatures=(Signatures)

#NewRealMatrix1=RandomSamplingMean(RealMatrix1)
#NewRealMatrix2=RandomSamplingMean(RealMatrix2)
#NewRealMatrix3=RandomSamplingMean(RealMatrix3)
#NewSignatures=(Signatures)

library("DeconRNASeq")

input1<-inputrim(NewRealMatrix1,NewSignatures)
input2<-inputrim(NewRealMatrix2,NewSignatures)
input3<-inputrim(NewRealMatrix3,NewSignatures)
dim(input1$NewRealMatrix)
dim(input2$NewRealMatrix)
dim(input3$NewRealMatrix)
```

```{r DeconvolutionRNAseq, echo=T}
Rlt1<-DeconRNASeq(input1$NewRealMatrix,input1$NewSignatures,checksig=FALSE,known.prop = F, use.scale = TRUE, fig = TRUE)
Rlt2<-DeconRNASeq(input2$NewRealMatrix,input2$NewSignatures,checksig=FALSE,known.prop = F, use.scale = TRUE, fig = TRUE)
Rlt3<-DeconRNASeq(input3$NewRealMatrix,input3$NewSignatures,checksig=FALSE,known.prop = F, use.scale = TRUE, fig = TRUE)
```

```{r}
acc1<-Rlt1$out.all
acc2<-Rlt2$out.all
acc3<-Rlt3$out.all

rownames(acc1)=colnames(NewRealMatrix1)
rownames(acc2)=colnames(NewRealMatrix2)
rownames(acc3)=colnames(NewRealMatrix3)

acc1<-data.frame(acc1)
acc2<-data.frame(acc2)
acc3<-data.frame(acc3)


gginput1<-data.frame(Conv=c(acc1[,2],acc2[,2]),group=c(rep("CCP",nrow(acc1)),rep("NP",nrow(acc2))))
gginput2<-data.frame(Conv=c(acc3[,8],acc2[,8]),group=c(rep("LCP",nrow(acc3)),rep("NP",nrow(acc2))))
#pdf("deconv.boxplot-compare.Full.contribution.pdf")
par(mfrow=c(2,2))
ggplot(gginput1,aes(factor(group), Conv))+geom_boxplot(outlier.shape = NA,aes(fill = "blue"))+coord_cartesian(ylim=c(0,0.15))+theme_bw()+theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank())+ geom_jitter()
ggplot(gginput2,aes(factor(group), Conv))+geom_boxplot(outlier.shape = NA,aes(fill = "blue"))+coord_cartesian(ylim=c(0,0.15))+theme_bw()+theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank())+ geom_jitter()
#dev.off()

# CCP vs NP
wilcox.test(acc1[,2],acc2[,2])
# LCP vs NP
wilcox.test(acc3[,8],acc2[,8])

# Colon cancer plasma deconvolution: frequency of DNA from colon cancer
sum(acc1[,2]>0)/nrow(acc1)
sum(acc2[,2]>0)/nrow(acc2)
# Lung cancer plasma deconvolution: frequency of DNA from colon cancer
sum(acc3[,8]>0)/nrow(acc3)
sum(acc2[,8]>0)/nrow(acc2)

# As our prior knownlege, WBC should be have higher proportion in the plasma (suppose>0.3)
acc1<-subset(acc1,WBC>0.3)
acc2<-subset(acc2,WBC>0.3)
acc3<-subset(acc3,WBC>0.3)

ci95(c(acc1[,12],acc2[,12],acc3[,12]))

ci95(c(acc1[,2]+acc1[,3]))
ci95(c(acc3[,8]+acc3[,10]))

ci95(c(acc1[,2],acc3[,8]))
ci95(c(acc1[,3],acc3[,10]))

ci95(c(acc2[,2]))
ci95(c(acc2[,8]))




acc1
boxplot(acc1,cex.axis=0.75)
colMeans(acc1)
acc2
boxplot(acc2,cex.axis=0.75)
colMeans(acc2)
acc3
boxplot(acc3,cex.axis=0.75)
colMeans(acc3)

out<-rbind(colMeans(acc1),colMeans(acc2),colMeans(acc3))
rownames(out)<-c("CCP","NP","LCP")
out

write.table(acc1,file="Colon-RealSample.Deconvolution.txt",col.names=NA,row.names=T,sep="\t",quote=F)
write.table(acc2,file="NormalPlasma-RealSample.Deconvolution.txt",col.names=NA,row.names=T,sep="\t",quote=F)
write.table(acc3,file="LungPlasma-RealSample.Deconvolution.txt",col.names=NA,row.names=T,sep="\t",quote=F)
write.table(colMeans(acc1),file="Colon-RealSample.ColMeans.Deconvolution.txt",col.names=NA,row.names=T,sep="\t,quote=F")
write.table(colMeans(acc2),file="NormalPlasma-RealSample.ColMeans.Deconvolution.txt",col.names=NA,row.names=T,sep="\t",quote=F)
write.table(colMeans(acc3),file="LungPlasma-RealSample.ColMeans.Deconvolution.txt",col.names=NA,row.names=T,sep="\t",quote=F)

# pdf("deconv.boxplot.contribution.pdf")
boxplot(acc1,cex.axis=0.7,ylim=c(0,1),horizontal = T,las=1,main="CCP")
boxplot(acc3,cex.axis=0.7,ylim=c(0,1),horizontal = T,las=1,main="LCP")
boxplot(acc2,cex.axis=0.7,ylim=c(0,1),horizontal = T,las=1,main="NP")
# dev.off()

gginput1<-data.frame(Conv=c(acc1[,2],acc2[,2]),group=c(rep("CCP",nrow(acc1)),rep("NP",nrow(acc2))))
gginput2<-data.frame(Conv=c(acc3[,8],acc2[,8]),group=c(rep("LCP",nrow(acc3)),rep("NP",nrow(acc2))))
#  pdf("deconv.boxplot-compare.contribution.pdf")
par(mfrow=c(2,2))
ggplot(gginput1,aes(factor(group), Conv))+geom_boxplot(outlier.shape = NA)+coord_cartesian(ylim=c(0,0.2))+theme_bw()+theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank())+ geom_jitter()
ggplot(gginput2,aes(factor(group), Conv))+geom_boxplot(outlier.shape = NA)+coord_cartesian(ylim=c(0,0.2))+theme_bw()+theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank())+ geom_jitter()
# dev.off()

ci95(c(acc1[,12],acc2[,12],acc3[,12]))

write.table(out,file="Plasma-RealSample.Same.Contribution.Deconvolution.txt",col.names=NA,row.names=T,sep="\t",quote=F)
save.image(file="Deconvolution.image.RData")
```

# Conclustion

We sought to quantify the tumor load based on the deconvolution analysis to the cancer and normal plasma by non-negative decomposition with quadratic programming with the reference of LC, CRC and 10 normal tissues. We estimated that a predominant fraction, 72.0% (95% CI:0.659-0.782) in the cancer and normal plasma were contributed by white blood cells, which is consistent with the levels reported in a recent study based on lower-coverage whole genome bisulfite sequencing (69.4%)16.  Primary tumor and normal tissue-of-origin contributes at the similar level of 2.3% (95% CI: 0.4%-4.2%) and 3.0% (95% CI:1.2%-4.4%). We also applied the similar analysis to normal plasma, and we found the tumor contribution to normal plasma were significantly lower (P=3.4x10-5 and 5.2x10-10 for CRC and LC, respectively) than that to cancer plasma, with the contribution of 0.17% and 1.0% for colon and lung cancer respectively. We also found 76.7% CRC and 89.6% LC samples were detected the contribution from tumor tissues while only 13% and 26% normal plasmas have certain (low) tumor contribution. Therefore, circulating cell-free DNA contains a relatively stable fraction of molecules released from various normal tissues, whereas in cancer patient tumor cells and apoptosis cells released DNA molecules that can be more abundant than normal tissues (Table S6). 

```{r closet,echo=T}
# R -e 'library("rmarkdown");library("knitr");rmarkdown::render("Deconv.Rmd")'
# R -e 'library("markdown");rpubsUpload("normalDev","Deconv.html")'
```


